src/sentry/static/sentry/app/views/eventsV2/results.tsx

```diff
@@ -80,18 +80,19 @@ class Results extends React.Component<Props, State> {
   componentDidUpdate(prevProps: Props, prevState: State) {
     const {api, location, organization, selection} = this.props;
     const {eventView} = this.state;
-    if (
-      !isEqual(prevProps.selection.projects, selection.projects) ||
-      !isEqual(prevProps.selection.datetime, selection.datetime)
-    ) {
-      loadOrganizationTags(api, organization.slug, selection);
-    }
 
     this.checkEventView();
     const currentQuery = eventView.getEventsAPIPayload(location);
     const prevQuery = prevState.eventView.getEventsAPIPayload(prevProps.location);
     if (!isAPIPayloadSimilar(currentQuery, prevQuery)) {
+      api.clear();
       this.fetchTotalCount();
+      if (
+        !isEqual(prevQuery.statsPeriod, currentQuery.statsPeriod) ||
```

Comment:
Not totally sure about this, but switched from prevProps.selection to put this inside the query check so that clear can be called before both updates. 

Otherwise `componentDidUpdate` updates twice and location vs. selection are out of sync, which causes `api.clear` to be called again, which then clears in-flight requests that we want.

Reply:
Dang, so they aren't equivocal then. Any prior art on how we can get around two different prop update checks clearing the api?

We could use multiple api instances but that feels hacky.

Reply:
Nvm, I see what you meant now, I had to choose an absolute date for it to appear in the query. 👍

Reply:
I don't know if there is much in the way of prior art. One potential option is to move this request into a wrapping component/child component that can better manage its props changes. The action that is called here populates a reflux store which is used by the search autocomplete on the page. As long as the tags are updated as the date + project selection changes the page will continue doing the right thing.

Reply:
I think as long as the selection/location props are being passed down to anything and they aren't update in the same tick you'll run into clearing problems. That or you do something along the lines of grouping or tagging api calls and clear by the tag.

Reply:
Anyway, I pushed an update to have it check start/end changes as well, still via query to avoid the api.clear issue for now.